File: est datos/python/selection_sort.py
def selection(arr):
    for i in range(0, len(arr)):
        peque = i
        for j in range(i+1, len(arr)):
            if arr[peque] > arr[j]:
                peque = j
        arr[i], arr[peque] = arr[peque], arr[i]
        printarr(arr)

def printarr(arr):
    for i in range(len(arr)):
        print(arr[i], end = " ")

    print('\n')

arr = [70,15,2,51,60,32]

print("Arreglo sin ordenar")
printarr(arr)
selection(arr)
print("\nArreglo ordenado")
printarr(arr)

File: est datos/cpp/selection_sort.cpp
#include <iostream>
#include <utility>

using namespace std;

void imprimir(int arr[]){
    for (int i= 0; i < 5; i++){
        cout << arr[i] << " ";
    }
}

void seleccion(int (&arr)[]){
    for( int i = 0; i < 5; i++){
        int peque = i;

        for (int j= i+1; j < 5; j++){
            if (arr[peque] > arr[j]){
                peque = j;
            }
        }
        swap(arr[i], arr[peque]);
        imprimir(arr);
        cout << '\n'; 
    }

}




int main(int argc, char const *argv[])
{
    int arr[] = {34,76,3,23,6};

    cout << "Arreglo sin ordenar: \n";
    imprimir(arr);
    seleccion(arr);
    cout << "\nArreglo ordenado: \n";
    imprimir(arr);
    return 0;
}